### ISSUE TITLE
Persist the container even if wait fails

### ISSUE THREAD

**Issue by leeederek** [kurtosis-tech/kurtosis#651]
### What's your CLI version?

0.77.0

### Description & steps to reproduce

Gyani - While working with AaronB - from Avalanche we were trying to play with `wait` on PortSpec as I had set it to None to get it to pass. I tried changing it to the default but that failed for the staking port (9651)

The problem is that when this failure happens the container gets deleted so we can't see logs. We can't talk to the container / get into it to see what the state of the process actually is.

### Desired behavior

Container remains even if the wait fails

### What is the severity of this bug?

Critical; I am blocked and Kurtosis is unusable for me because of this bug.

**Comment by mieubrisse** [comment]
Logs that @gbouv is doing might help here

**Comment by gbouv** [comment]
I agree this is a huge pain. I filed this a while ago: https://github.com/kurtosis-tech/kurtosis-private/issues/1168
I cannot count the number of time I had to use a custom-build of kurtosis to work around the "auto-clean" behaviour when something fails.
I am not sure the keeping the logs would address 100% of the issue here, but I agree it would makes things slightly better, provided that we have enough time to capture those logs before the container is removed

**Comment by h4ck3rk3y** [comment]
Ideally I want more than logs. Ideally I want to ssh into the container and see whats going on at that point in time.

**Comment by h4ck3rk3y** [comment]
My flow around a lot of these; `Kurtosis removes everything if failure` bugs has been to put in a `sleep 999999` and then start the container. Like this morning -

I had the path to a config file off and the logs (via docker ps -a) would tell me `/foo/bar/doo.json` file not found while within Kurtosis the service didn't exist as the service never started. 

The `docker ps -a` flow is nice(to tell me what the logs) but even then I can't ssh into that container as that container has exited.

I'm coupling a few things here but

1. We shouldn't lose logs
2. For ports that don't respond within time out - we shouldn't delete the container. I still want to `ssh` into this

For service's that just can't startup (wrong config file). I'd like to be able to ssh into it but thats a nice to have.

**Comment by gbouv** [comment]
I was thinking about it yesterday evening, and I am considering the option of just removing the "clean all if failure" behaviour. I think this kind of behaviour could make sense in a world where we capture enough state, but it seems we're not meeting the bar right now, and giving I fee leaving the enclave in a weird state is better to debug than removing everything and leaving the user with nothing to look at.

**Comment by mieubrisse** [comment]
I think this is probably right, for now. We really should be capturing that state though, so I'd prefer if we can mark it as a hack
